kinbot/mess.py

```diff
@@ -144,127 +167,316 @@ def write_input(self):
+                            prod_fr_iter.append(prod_fr)
+                        elif len(reaction.products) == 2:
+                            bimol_name = '_'.join(sorted([str(st_pt.chemid) for st_pt in reaction.products]))
+                            bimolec_blocks[bimol_name], bimol_e, bimol_fr = self.write_bimol([opt.species for opt in reaction.prod_opt],
+                                                                                             0,
+                                                                                             uq,
+                                                                                             uq_n,
+                                                                                             lenProd,
+                                                                                             energyAdd,
+                                                                                             freqFactor,
+                                                                                             uq_iter)
+                            bimol_e_iter.append(bimol_e)
+                            bimol_fr_iter.append(bimol_fr)
+                        else:
+                            # termol
+                            termolec_ts_blocks[reaction.instance_name] = allTS[reaction.instance_name]
+                            termol_name = '_'.join(sorted([str(st_pt.chemid) for st_pt in reaction.products]))
+                            termolec_blocks[termol_name] = self.write_termol([opt.species for opt in reaction.prod_opt], reaction, uq, uq_n, energyAdd, freqFactor, 0, uq_iter)
+
+                # Homolytic scission - barrierless reactions
+                barrierless = {}
+                if self.species.homolytic_scissions is not None:
+                    for hs in self.species.homolytic_scissions.hss:
+                        bar = 1
+                        barrierless_freqFactor = uq_obj.calc_freqUQ(freq_uqVal)
+                        barrierless_energyAdd = uq_obj.calc_energyUQ(well_uqVal)
+                        new = 1
+                        if hs.status == -1:
+                            prod_name = '_'.join(sorted([str(prod.chemid) for prod in hs.products]))
+                            if prod_name in self.barrierless_names:
+                                new = 0
+                            if new:
+                                self.barrierless_names[prod_name] = prod_name
+                            if new == 1 or uq_iter >= 1:
+                                barrierless_blocks[prod_name], barrierless_e, barrierless_fr = self.write_barrierless([opt.species for opt in hs.prod_opt],
+                                                                                                                      hs,
+                                                                                                                      uq,
```

Comment:
this is too much space here

Reply:
I will fix that. I think I had previously had some post processing lines there, which I deleted to move them to the 'uncertaintyAnalysis.py' file